test/models/name_test.rb

```diff
@@ -1297,6 +1297,22 @@ def test_name_prov_name_with_spaces
     )
   end
 
+  def test_name_prov_name_with_trailing_nonbreaking_space
+    do_name_parse_test(
+      "Cuphophyllus \"pratensis-IN01\"\u00A0",
```

Comment:
I'd include a leading nonbreaking space as well and perhaps even one between the genus and the provisional species name if you take my above recommendation.